bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -661,6 +693,11 @@ void checkAllLedgers() throws BKException, IOException, InterruptedException, Ke
                     }
                 }, null, BKException.Code.OK, BKException.Code.ReadException);
             FutureUtils.result(processFuture, BKException.HANDLER);
+            try {
+                ledgerUnderreplicationManager.setCheckAllLedgersCTime(System.currentTimeMillis());
+            } catch (UnavailableException ue) {
+                LOG.error("Got exception while trying to set checkAllLedgersCTime", ue);
```

Comment:
Since there is a potential to fail this, why not cache this value. and the auditor reads the zk only if the local/cached value is -1. A minor optimization and also saves us from this kind of errors.

Reply:
didn't get you completely. Also not sure how cache would work in all the scenarios here.

Anyhow frequency of checkAllLedgers is quite low (once a week), I don't see much value in this minor optimization and not sure if it is worth complicating logic.